demo/public/javascripts/demo.js

```diff
@@ -0,0 +1,87 @@
+(function () {
+    let otherLoads;
+
+    if (window.onload) {
+        otherLoads = window.onload;
+    }
+
+    let updateBackground = function () {
+        const el = document.getElementById("tsparticles");
+        const options = tsParticles.domItem(0).options;
+        const config = options.config_demo;
+        let backgroundImage;
+
+        if (config.background_image) {
+            if (config.background_image.startsWith("url(")) {
+                backgroundImage = config.background_image;
+            } else {
+                backgroundImage = `url('${config.background_image}')`;
+            }
+        } else {
+            backgroundImage = '';
+        }
+
+        el.style.backgroundColor = config.background_color;
+        el.style.backgroundImage = backgroundImage;
+        el.style.backgroundPosition = config.background_position;
+        el.style.backgroundRepeat = config.background_repeat;
+        el.style.backgroundSize = config.background_size;
+    };
+
+    let updateParticles = function (editor) {
+        let presetId = localStorage.presetId || 'default';
+
+        tsParticles.loadJSON('tsparticles', `/presets/${presetId}.json`).then((particles) => {
+            localStorage.presetId = presetId;
+            editor.set(particles.options);
+            editor.expandAll();
+            updateBackground();
+        });
+    };
+
+    window.onload = function () {
+        if (otherLoads) {
+            otherLoads();
+        }
+
+        const element = document.getElementById('editor');
+        const options = {
+            mode: 'tree',
+            modes: ['code', 'form', 'text', 'tree', 'view', 'preview'], // allowed modes
+            onError: function (err) {
+                alert(err.toString())
+            },
+            onModeChange: function (newMode, oldMode) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['newMode' is defined but never used.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5136254)